packages/common/src/store/gated-content/sagas.ts

```
@@ -423,94 +430,110 @@ export function* pollGatedTrack({
     DEFAULT_GATED_TRACK_POLL_INTERVAL_MS
 
   // get initial track metadata to determine whether we are polling for stream or download access
-  const cachedTracks = yield* select(getTracks, {
-    ids: [trackId]
-  })
-  const initialTrack = cachedTracks[trackId]
-  const initiallyHadNoStreamAccess = !initialTrack?.access.stream
-  const initiallyHadNoDownloadAccess = !initialTrack?.access.download
+  const isAlbum = contentType === 'album'
+  const cachedEntity = isAlbum
+    ? yield* select(getCollection, { id: contentId })
+    : yield* select(getTrack, {
+        id: contentId
+      })
+  const initiallyHadNoStreamAccess = !cachedEntity?.access.stream
+  const initiallyHadNoDownloadAccess = !cachedEntity?.access.download
 
   // poll for access until it is granted
   while (true) {
-    const track = yield* call([apiClient, 'getTrack'], {
-      id: trackId,
-      currentUserId
-    })
-    const currentlyHasStreamAccess = !!track?.access?.stream
-    const currentlyHasDownloadAccess = !!track?.access?.download
+    const apiEntity = isAlbum
+      ? (yield* call([apiClient, 'getPlaylist'], {
+          playlistId: contentId,
+          currentUserId
+        }))[0]
+      : yield* call([apiClient, 'getTrack'], {
+          id: contentId,
+          currentUserId
+        })
 
-    if (initiallyHadNoStreamAccess && currentlyHasStreamAccess) {
-      yield* put(
-        cacheActions.update(Kind.TRACKS, [
-          {
-            id: trackId,
-            metadata: {
-              access: track.access
-            }
+    if (!apiEntity?.access) {
```

we should probably throw an error here?